rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -186,14 +321,60 @@ components:
       maxLength: 80
       pattern: '^([1-9]([0-9]+)?)\.(0|([1-9]([0-9]+)?))(\.(0|([1-9]([0-9]+)?))(-.*)?)?$'
       example: 1.2, 2.4.3, 0.2.5-SNAPSHOT
+    MetadataInformationResponse:
+      type: object
+      required:
+        - transformationStatus
+        - metadataInformations
+      properties:
+        transformationStatus:
+          $ref: '#/components/schemas/Status'
+        errorMessage:
+          description: An optional error message in the case the transformation failed.
+          $ref: '#/components/schemas/ErrorMessage'
+        metadataInformations:
+          type: array
+          items:
+            $ref: '#/components/schemas/MetadataInformation'
+    MetadataInformation:
+      type: object
+      required:
+        - metadataType
+        - transformationStatus
+      properties:
+        metadataType:
+          $ref: '#/components/schemas/MetadataType'
+        transformationStatus:
+          $ref: '#/components/schemas/Status'
+        errorMessage:
+          description: An optional error message in the case the transformation failed.
+          $ref: '#/components/schemas/ErrorMessage'
+        location:
+          description: The URI for where the metadata is located. NOTE this will not be populated until the transformationStatus is Complete.
+          type: string
+          format: uri
+          example: 'https://www.example.com/mis/30f14c6c1fc85cba12bfd093aa8f90e3'
+        transformedTimestamp:
+          type: string
+          format: date-time
+          description: The server time when the metadata was generated. NOTE this will not be populated until the transformationStatus is Complete.
+          example: 2019-05-09T14:47:13.101+0000
+    MetadataType:
+      description: The type of metadata.
+      type: string
+      enum: [irm]
+    Status:
+      description: The status of the operation.
+      type: string
+      enum: [In Progress, Done, Failed]
     TransformRequest:
       type: object
       required:
         - location
         - mimeType
         - metacardLocation
       properties:
-        location:
+        currentLocation:
```

Comment:
✏️ Maybe add a note that this needs to be active until the DELETE request (or maybe the status says DONE)

Reply:
Would you prefer we (eventually) provide a status in the transform response that indicates it is no longer required? I am asking because technically, once we download it and start making the determination of what we will do with it, we no longer need it.

Reply:
If you only download it once, why do we send the URI to you instead of the actual file?

Reply:
very good question

Reply:
The more I think about it, if transform ever has a need to restart a transformation from scratch because it died in the middle, it might have a need to re-retrieve the file. In addition, there are potentially more than one file being exchanged, for example, right now there is both the product and one type of metadata - the metacard and we might have more later so, keeping it with URIs supports these cases. Finally, until we support a better way than hardcoding filenames as we did on the ingest API for rep/sync, I would want to stay away from that.

Reply:
lol ... but ... now that transform will have it's local repository to hold data, it could always download it and store it there and only require the info for up to the point where its state goes to TRANSFORMING